src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -223,6 +221,89 @@ class RefreshIndexTests extends QueryTest with HyperspaceSuite {
+      "expected, when some file gets deleted and some appended to source data.") {
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      withIndex(indexConfig.indexName) {
+        SampleData.save(
+          spark,
+          nonPartitionedDataPath,
+          Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+        val df = spark.read.parquet(nonPartitionedDataPath)
+        hyperspace.createIndex(df, indexConfig)
+
+        // Delete one source data file.
+        deleteDataFile(nonPartitionedDataPath)
+
+        val oldFileList = fileList(nonPartitionedDataPath).toSet
+
+        // Add some new data to source.
+        import spark.implicits._
+        SampleData.testData
+          .take(3)
+          .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+          .write
+          .mode("append")
+          .parquet(nonPartitionedDataPath)
+
+        hyperspace.refreshIndex(indexConfig.indexName)
+
+        // Check if refreshed index metadata has "appendedFiles" and "deletedFiles" updated.
+        val entry = logManager(indexConfig.indexName).getLatestStableLog()
+        assert(entry.isDefined)
+        assert(entry.get.isInstanceOf[IndexLogEntry])
+        val indexLogEntry = entry.get.asInstanceOf[IndexLogEntry]
+        assert(indexLogEntry.deletedFiles.isEmpty)
+
+        val newFileList = fileList(nonPartitionedDataPath).toSet
+        assert(indexLogEntry.appendedFiles.toSet.equals(newFileList -- oldFileList))
+      }
+    }
+  }
+
+  test(
+    "Validate refresh append action updates appended and deleted files in metadata as" +
+      "expected, when some file gets deleted and some appended to source data.") {
+    withTempPathAsString { testPath =>
+      withSQLConf(IndexConstants.REFRESH_APPEND_ENABLED -> "true") {
+        withIndex(indexConfig.indexName) {
+          SampleData.save(spark, testPath, Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+          val df = spark.read.parquet(testPath)
+          hyperspace.createIndex(df, indexConfig)
+
+          val oldFileList = fileList(testPath).toSet
+
+          // Delete one source data file.
+          deleteDataFile(testPath)
+
+          // Add some new data to source.
+          import spark.implicits._
+          SampleData.testData
+            .take(3)
+            .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+            .write
+            .mode("append")
+            .parquet(testPath)
+
+          hyperspace.refreshIndex(indexConfig.indexName)
+
+          // Check if refreshed index metadata has "appendedFiles" and "deletedFiles" updated.
+          val entry = logManager(indexConfig.indexName).getLatestStableLog()
+          assert(entry.isDefined)
+          assert(entry.get.isInstanceOf[IndexLogEntry])
+          val indexLogEntry = entry.get.asInstanceOf[IndexLogEntry]
+          assert(indexLogEntry.appendedFiles.isEmpty)
```

Comment:
Can you also assert `latestFiles -- oldFiles` is empty?

Reply:
thanks, @imback82 , added